### ISSUE TITLE
ng-content gets created when its declaring component gets created even when in a template that is not created

### ISSUE THREAD

**Issue by davityle** [angular/angular#16550]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
`ng-content` gets created when its declaring component gets created even when in a template that is not created. The `ng-content` also does not get destroyed or recreated with the template that it is declared in when the template is destroyed or recreated.

**Expected behavior**
`ng-content` should be created and destroyed with the template that it is declared in. 

**Minimal reproduction of the problem with instructions**
https://plnkr.co/edit/jX9909yNaic5DeElY6HJ?p=preview

**What is the motivation / use case for changing the behavior?**
Not having things created when they shouldn't be created and having things get destroyed - not just hidden - when they should be destroyed.

**Please tell us about your environment:**

* **Angular version:** 2.0.X
Angular 4.1.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` =

**Comment by zoechi** [comment]
AFAIK this is as designed. Even when the content is not transcluded, you can still query it using `@ContentChild()` or `@ContentChildren()`

**Comment by zoechi** [comment]
If you want content only to be created when it is stamped, you can pass a template like 

```ts
<wrapper-component>
  <ng-template>
    <component-that-should-never-get-created></component-that-should-never-get-created>
  </ng-template>
</wrapper-component>
```
and then use `ngTemplateOutlet` or `ngForTemplate` to stamp the template once or multiple times.

**Comment by davityle** [comment]
Why is it designed that way though? Why do I need to create a second template to keep a component from getting created when it is already in an `*ngIf`? The component we found this in was a collapsible component that looked like ```<header></header>
<div *ngIf="!collapsed" class="content"><ng-content></ng-content></div>```. We expected that the content of this component would not be created until the template that contained the content was created. It doesn't seem correct to me that the components would live in memory when they don't actually live on the document.

**Comment by zoechi** [comment]
The component is not in an `*ngIf` only the maker element that indicates where it should be transcluded.
`<ng-content>` is not about whether it's created or not, it's **only** about **where** it is displayed.
`<wrapper-component><component-that-should-never-get-created></component-that-should-never-get-created></wrapper-component>` is where it is created.

**Comment by devoto13** [comment]
More and more people seems to be affected by lack of documentation on this... https://github.com/angular/angular.io/issues/3099

**Comment by davityle** [comment]
@zoechi I updated our components to use templates and since we are using `ChangeDetectionStrategy.OnPush` we ran into this bug https://github.com/angular/angular/issues/16012 . The workarounds for that are fairly hacky. So my understanding of my choices right now are:

a) allow components to take memory when they aren't actually in use along with not following conventional lifecycle hooks when added or removed from the dom (which isn't actually a choice because we have some heavy components hidden in this way and we depend on reliable lifecycle hooks)

b) perform some brittle or widely scoped hacky workarounds while using templates

c) rework the components so that the showing and hiding of the content is handled in the same component that declares the content which would involve a lot of very repetitive boilerplate. 

Can you think of a better way?

**Comment by zoechi** [comment]
I didn't run into this issue myself yet and don't know details. Perhaps someone else.

**Comment by davityle** [comment]
It looks like https://github.com/angular/angular/issues/12816 has been fixed which makes our workaround for https://github.com/angular/angular/issues/16012 much cleaner. So b it is. (thanks @tbosch :tada:)

**Comment by tbosch** [comment]
@davityle there is also a fix for #16012 now in https://github.com/angular/angular/pull/16592. Will be landed soon.

Closing this issue as the behavior of `ng-content` is works as expected.
